[e2e tests]: fix test 6987 in operator suite

enp0s3 opened the pull request:
Signed-off-by: Igor Bezukh <ibezukh@redhat.com>

**What this PR does / why we need it**:
The test aims to verify that virt-handler is being updated
with the configured maximum amount of devices. The test was
missing the validation of Kubevirt status, whether deployment of new
configuration has finished. It also missed checking that all the virt-handler
pods are ready after the rollout.

This caused flakes in the subsequent tests i.e. `test_id:4744` was failed on timeout waiting for the patch 
to be propogated.

**Release note**:
```release-note
NONE
```

dhiller commented on tests/operator_test.go:
```diff
@@ -1374,9 +1374,15 @@ spec:
 			Eventually(fetchVirtHandlerCommand, 60*time.Second, 5*time.Second).Should(ContainSubstring(maxDevicesCommandArgument))
 
+			By("Waiting for virt-handler DS to become ready")
+			verifyVirtHandlerIsReady(virtClient, kv, 60)
```
nit: I see a contradiction here of what the `By` says to what the name of the function tells me. Maybe we should rename the function to be similar to `waitForKvWithTimeout`, i.e. `waitForVirtHandlerReadyWithTimeout`. WDYT?

enp0s3 replied:
@dhiller Thanks! fixed

xpivarc commented on tests/operator_test.go:
```diff
@@ -1385,8 +1391,14 @@ spec:
 			kv, err = virtClient.KubeVirt(flags.KubeVirtInstallNamespace).Update(kv)
 			Expect(err).ToNot(HaveOccurred())
 
+			By("Waiting for virt-handler component to be patched")
```
Same as above.

xpivarc commented on tests/operator_test.go:
```diff
@@ -1374,9 +1374,15 @@ spec:
 			By("Test that patch was applied to DaemonSet")
 			Eventually(fetchVirtHandlerCommand, 60*time.Second, 5*time.Second).Should(ContainSubstring(maxDevicesCommandArgument))
 
+			By("Waiting for virt-handler DS to become ready")
```
Why do we need this?

dhiller reviewed (commented):
Thanks!

I am not unholding for now, because Lubo still has questions.

enp0s3 commented:
@xpivarc Hi, can you PTAL? I've addressed your comments, now I will just wait for the right KV phase and conditions, I think I should rely only on that, and not mimic the validation that the operator does itself. I've made local tests and there is not such situation where the KV status will flip-flop from `Available` to `Deploying` while making changes to a component.

enp0s3 commented:
@xpivarc Also, I would like to refactor the [`EnsureKubevirtInfra()` ](https://github.com/kubevirt/kubevirt/blob/c4b6ae63c5a7f642ab86b0755dabca3b814ecb39/tests/testsuite/fixture.go#L138) function to poll the KV status and eventually assert on it, instead of replicating the validation code from the operator. I think we should rely on the reported statuses and fix them if necessary.

enp0s3 commented:
/test pull-kubevirt-e2e-k8s-1.22-operator

enp0s3 commented:
suggested timeouts seem to work : https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/8083/pull-kubevirt-e2e-k8s-1.22-operator/1549335550362652672#1:build-log.txt%3A2247

xpivarc reviewed (commented):
/approve
Lets see if we don't going to need some consistent check

xpivarc commented:
/lgtm

xpivarc commented:
/retest

enp0s3 commented:
/unhold
